BUILDING.md

```diff
@@ -31,4 +39,78 @@ The following trivial build script changes were made to successfully run `./grad
 
-We plan to remove the need for this work-around via [OpenSearch#581](https://github.com/opensearch-project/OpenSearch/issues/581).
\ No newline at end of file
+We plan to remove the need for this work-around via [OpenSearch#581](https://github.com/opensearch-project/OpenSearch/issues/581).
+
+## Developing new Plugins for OpenSearch
+
+All OpenSearch Plugins must contain the `plugin-descriptor.properties` file. The fields contained in the `plugin-descriptor.properties` file are mentioned [here](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/resources/plugin-descriptor.properties). If using the gradle build system, these fields can be specified in the `build.gradle` file and the build system will create the properties file for us.
+
+- [](#)
+  - [Building Plugins with OpenSearch](#building-plugins-with-opensearch)
+    - [Publish OpenSearch to Maven Local](#publish-opensearch-to-maven-local)
+    - [Use OpenSearch from Maven Local in Plugins](#use-opensearch-from-maven-local-in-plugins)
+  - [Developing new Plugins for OpenSearch](#developing-new-plugins-for-opensearch)
+    - [Example `plugin-descriptor.properties` file](#example-plugin-descriptorproperties-file)
+    - [Example `build.gradle` file](#example-buildgradle-file)
+    - [Custom folder name for Plugins](#custom-folder-name-for-plugins)
+    - [Notes](#notes)
+
+### Example `plugin-descriptor.properties` file
+
+Following is a list of required and optional elements for the Plugins:
+
+```
+### mandatory elements for all plugins:
+#
+# 'description': simple summary of the plugin
+description=${description}
+#
+# 'version': plugin's version
+version=${version}
+#
+# 'name': the plugin name
+name=${name}
+#
+# 'classname': the name of the class to load, fully-qualified.
+classname=${classname}
+#
+# 'java.version': version of java the code is built against
+# use the system property java.specification.version
+# version string must be a sequence of nonnegative decimal integers
+# separated by "."'s and may have leading zeros
+java.version=${javaVersion}
+#
+# 'opensearch.version': version of opensearch compiled against
+opensearch.version=${opensearchVersion}
+### optional elements for plugins:
+#
+# 'custom.foldername': the custom name of the folder in which the plugin is installed.
+custom.foldername=${customFolderName}
+#
+#  'extended.plugins': other plugins this plugin extends through SPI
+extended.plugins=${extendedPlugins}
+#
+# 'has.native.controller': whether or not the plugin has a native controller
+has.native.controller=${hasNativeController}
+```
+
+### Example `build.gradle` file
+
+If adding the properties in `build.gradle` file, below is an example for the `opensearchplugin`:
+
+```
+opensearchplugin {
+    name 'opensearch-plugin'
+    description 'Example OpenSearch plugin'
+    classname 'org.opensearch.example.Plugin'
+}
+```
+### Custom folder name for Plugins
```

Comment:
It's strange that we only document the custom folder name here, especially since for all the other fields you refer people elsewhere. So either add a table with supported fields here and an explanation, or point to a place with docs (javadoc?) that documents them, including for `customFolderName`.

Reply:
I mentioned the custom folder name since its a new feature. But you are right, the title of the section should be different, something that represents new features. What do you think?

Reply:
I would document current state, and not worry about what's new and what's not new since everything gets old fast.

Reply:
Makes sense! Updated it. Also added a table to include all the elements.